pallets/stream-payment/src/lib.rs

```diff
@@ -0,0 +1,936 @@
+        /// Payee, target of the stream.
+        pub target: AccountId,
+        /// Steam config (time unit, asset id, rate)
+        pub config: StreamConfig<Unit, AssetId, Balance>,
+        /// How much is deposited to fund this stream.
+        pub deposit: Balance,
+        /// Last time the stream was updated in `config.time_unit`.
+        pub last_time_updated: Balance,
+        /// Nonce for requests. This prevents a request to make a first request
+        /// then change it to another request to frontrun the other party
+        /// accepting.
+        pub request_nonce: RequestNonce,
+        /// A pending change request if any.
+        pub pending_request: Option<ChangeRequest<Unit, AssetId, Balance>>,
+    }
+
+    impl<AccountId: PartialEq, Unit, AssetId, Balance> Stream<AccountId, Unit, AssetId, Balance> {
+        pub fn account_to_party(&self, account: AccountId) -> Option<Party> {
+            match account {
+                a if a == self.source => Some(Party::Source),
+                a if a == self.target => Some(Party::Target),
+                _ => None,
+            }
+        }
+    }
+
+    /// Stream configuration.
+    #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+    #[derive(RuntimeDebug, PartialEq, Eq, Encode, Decode, Copy, Clone, TypeInfo)]
+    pub struct StreamConfig<Unit, AssetId, Balance> {
+        /// Unit in which time is measured using a `TimeProvider`.
+        pub time_unit: Unit,
+        /// Asset used for payment.
+        pub asset_id: AssetId,
+        /// Amount of asset / unit.
+        pub rate: Balance,
+    }
+
+    /// Origin of a change request.
+    #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+    #[derive(RuntimeDebug, PartialEq, Eq, Encode, Decode, Copy, Clone, TypeInfo)]
+    pub enum Party {
+        Source,
+        Target,
+    }
+
+    impl Party {
+        pub fn inverse(self) -> Self {
+            match self {
+                Party::Source => Party::Target,
+                Party::Target => Party::Source,
+            }
+        }
+    }
+
+    /// Kind of change requested.
+    #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+    #[derive(RuntimeDebug, PartialEq, Eq, Encode, Decode, Copy, Clone, TypeInfo)]
+    pub enum ChangeKind<Time> {
+        /// The requested change is a suggestion, and the other party doesn't
+        /// need to accept it.
+        Suggestion,
+        /// The requested change is mandatory, and the other party must either
+        /// accept the change or close the stream. Reaching the deadline will
+        /// close the stream too.
+        Mandatory { deadline: Time },
+    }
+
+    /// Describe how the deposit should change.
+    #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+    #[derive(RuntimeDebug, PartialEq, Eq, Encode, Decode, Copy, Clone, TypeInfo)]
+    pub enum DepositChange<Balance> {
+        Increase(Balance),
+        Decrease(Balance),
+        Absolute(Balance),
```

Comment:
I think it Absolute means that such balance will be applied as a rate, but it would be good small comments in each enum variant